internal/zinc-core/src/main/scala/sbt/internal/inc/Stamp.scala

```diff
@@ -58,10 +59,10 @@ trait WithPattern { protected def Pattern: Regex }
 import java.lang.{ Long => BoxedLong }
 
 /** Define the hash of the file contents. It's a typical stamp for compilation sources. */
-final class Hash(val value: Array[Byte]) extends StampBase {
-  val hexHash = IOHash.toHex(value)
+final class Hash private[inc] (val hexHash: String) extends StampBase {
```

Comment:
Either of those alternatives have a runtime performance penalty (returning `Option` and `Either` require validation), which goes against the goal of the commit that introduced this change.
  
IMO, the best to solve this issue is to put all those stamp utils in its own package, and then do `private[stamps]`.